metagpt/utils/serialize.py

```diff
@@ -35,14 +33,45 @@ def actionoutout_schema_to_mapping(schema: Dict) -> Dict:
         if property["type"] == "string":
             mapping[field] = (str, ...)
         elif property["type"] == "array" and property["items"]["type"] == "string":
-            mapping[field] = (List[str], ...)
+            mapping[field] = (list[str], ...)
         elif property["type"] == "array" and property["items"]["type"] == "array":
-            # here only consider the `List[List[str]]` situation
-            mapping[field] = (List[List[str]], ...)
+            # here only consider the `list[list[str]]` situation
+            mapping[field] = (list[list[str]], ...)
     return mapping
 
 
-def serialize_message(message: Message):
+def actionoutput_mapping_to_str(mapping: dict) -> dict:
+    new_mapping = {}
+    for key, value in mapping.items():
+        new_mapping[key] = str(value)
+    return new_mapping
+
+
+def actionoutput_str_to_mapping(mapping: dict) -> dict:
+    new_mapping = {}
+    for key, value in mapping.items():
+        if value == "(<class 'str'>, Ellipsis)":
+            new_mapping[key] = (str, ...)
+        else:
+            new_mapping[key] = eval(value)  # `"'(list[str], Ellipsis)"` to `(list[str], ...)`
+    return new_mapping
+
+
+def serialize_general_message(message: "Message") -> dict:
```

Comment:
is independent function for ser/deser message necessary?

Reply:
Message is a little different, in old version, dynamic BaseModel class like `prd` created by `create_model_class` in `instruct_content` can't be directly deserialized.
There two function will be rm due to not used in main code.